### PR TITLE
feat: address few bug with linting and improve highlighting

### LINKED ISSUE #6800: [Bug] Comment lines are being checked for JS code
## Description
Comment `//` and `/* */` line is being checked for JS syntax hence incorrect linting error is displayed.

### Steps to reproduce the behaviour:
1. Input comment line on JS code and include variables enclosed in quotes or using special character (see screenshot) and observe linting operation on comments

### Important Details
- Version: Cloud
- OS: Win10
- Browser: Chrome
- Environment: Deploy Preview (6265 Linting & Errors)

image

### LINKED ISSUE #6811: [Bug] Console log code shouldn't be displayed as warning/errors
## Description
Currently,  system picks up console as a keyword and shows an undefined warning. If code has `console.log` string then it must be omitted from lint check

### Important Details
- Version: Cloud
- OS: Win10
- Browser: Chrome
- Environment: Deploy Preview (6265 Linting & Errors)

image

### LINKED ISSUE #6900: [Bug] New line character breaks linting when used in console log
## Description
New line character or other escape characters are breaking linting on new and existing code

### Steps to reproduce the behaviour:
1. add an ecaspe character inside of a console log statement and observe linting break

### Important Details
- Version: Cloud
- OS: Win10
- Browser: Chrome
- Environment: Deploy preview (6265 Linting & Errors)

[LOOM DEMO](https://www.loom.com/share/758732df63c2445badbfe2ffbbcc0618)

### LINKED ISSUE #6901: [Bug] Linting is showing up on AppSmith API showModal(), closeModal() etc
## Description
Linting is showing up on AppSmith API showModal(), closeModal(), download() etc., when used on a tooltip field (or any other field that's not under 'Actions' of property pane)

### Steps to reproduce the behaviour:
1. Add showModal() as JS code in a tooltip field of an input widget and observe linting show up

### Important Details
- Version: Cloud
- OS: Win10
- Browser: Chrome
- Environment: Deploy Preview (6265 Linting & Errors)

image

### LINKED ISSUE #7163: Same errors are displayed on 2 different lines

### LINKED ISSUE #7170: Moment is displayed as warning

### LINKED ISSUE #7171: Incorrect error and location of error

### PR DESCRIPTION
**Pull request by vnodecg**
## Description

Addresses the following issues with linter

- [x] 1. Lint errors are not displayed until control is within the editor
- [x] 2. Disable all warnings
- [x] 3. Global data not getting imported
- [x] 4. For scanning/parsing issues - highlight an entire code block
- [x] 5. Same errors are displayed on 2 different lines
- [ ] 6. Typo on suggestion for nested properties (add multi level global data) // Not possible
- [x] 7. === issue (remove this rule)

We are using positions from jshint instead of string matching. This gives us much better location on errors

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?

Manually tested

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

Fixes #7170 
Fixes #7171 
Fixes #6901
Fixes #6811 
Fixes #6800 
Fixes #7163 
Fixes #6900 

## Test coverage results :test_tube:
<details><summary>:green_circle: Total coverage has increased</summary>

    // Code coverage diff between base branch:release and head branch: FEATURE/Show-lint-errors-in-codeeditor 
Status | File | % Stmts | % Branch | % Funcs | % Lines 
 -----|-----|---------|----------|---------|------ 
 :green_circle: | total | 54.89 **(0.01)** | 36.9 **(0.02)** | 34.22 **(-0.01)** | 55.41 **(0.01)**
 :green_circle: | app/client/src/components/editorComponents/CodeEditor/index.tsx | 67.91 **(0.12)** | 40 **(0.39)** | 52.5 **(0)** | 68.7 **(0.13)**
 :green_circle: | app/client/src/components/editorComponents/CodeEditor/lintHelpers.ts | 97.1 **(5.32)** | 76.47 **(-2.1)** | 100 **(0)** | 96.36 **(7.27)**
 :red_circle: | app/client/src/sagas/PostEvaluationSagas.ts | 17.36 **(-0.12)** | 0 **(0)** | 0 **(0)** | 21.19 **(-0.18)**
 :green_circle: | app/client/src/workers/evaluate.ts | 95.06 **(0.54)** | 85 **(5)** | 88.89 **(-0.58)** | 94.94 **(0.5)**</details>

### CODE REVIEW

File: app/client/src/workers/evaluate.ts [hunk 1]
```diff
@@ -8,9 +8,10 @@ import {
 import unescapeJS from "unescape-js";
 import { JSHINT as jshint } from "jshint";
 import { Severity } from "entities/AppsmithConsole";
+import { Position } from "codemirror";
 import { AppsmithPromise, enhanceDataTreeWithFunctions } from "./Actions";
 import { ActionDescription } from "entities/DataTree/actionTriggers";
-import { isEmpty } from "lodash";
+import _ from "lodash";
```
Comment (hetunandu) [inline app/client/src/workers/evaluate.ts:14]: Only import lodash functions that are needed

**Review by hetunandu — APPROVED** [review]

**Comment by ramsaptami** [comment]
@hetunandu @vnodecg not sure if the approach to highlight the entire block is an appropriate way to deal with parsing issues like seen in the screenshot below. User would be unable to pin point the issue if this was a longer code whereas the issue is to do with missing brackets. Our documentation and Google search aren't helpful in this case either.
image

**Comment by hetunandu** [comment]
> @hetunandu @vnodecg not sure if the approach to highlight the entire block is an appropriate way to deal with parsing issues like seen in the screenshot below. User would be unable to pin point the issue if this was a longer code whereas the issue is to do with missing brackets. Our documentation and Google search aren't helpful in this case either.
> image

If that is a case then lets not show any lint errors incase there is a parsing issue

**Comment by ramsaptami** [comment]
Makes sense for most parsing errors. But in the case of missing bracket pairs, can we still do something because that's a common error and also something most IDE's do for UX

**Review by hetunandu — APPROVED** [review]

**Review by hetunandu — APPROVED** [review]

**Pull request merged**